src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1928,94 +1947,217 @@ type reqFunc func(context.Context, interface{}) (interface{}, error)
 
 func (k *kataAgent) installReqFunc(c *kataclient.AgentClient) {
 	k.reqHandlers = make(map[string]reqFunc)
-	k.reqHandlers[grpcCheckRequest] = func(ctx context.Context, req interface{}) (interface{}, error) {
-		return k.client.HealthClient.Check(ctx, req.(*grpc.CheckRequest))
+
+	checkReq := func(handlerId string, k *kataAgent, req interface{}) (err error) {
+		defer errors.ErrorContext(&err, "Failed to handle request "+handlerId)
+		defer func() {
+			if err != nil {
+				if k.deadErr != nil {
+					context := err.Error()
+					err = k.deadErr
+					errors.ErrorContext(&err, context)
+				}
+			}
+
+		}()
+
+		if k == nil {
+			err = errors.Errorf("kataAgent is nil")
+			return
+		}
+
+		if k.client == nil {
+			err = errors.Errorf("kataAgent client is nil for " + handlerId)
+			return
+		}
+
+		if req == nil {
+			err = errors.Errorf("req is nil")
+			return
+		}
+		return nil
 	}
+
+	k.reqHandlers[grpcCheckRequest] = func(ctx context.Context, req interface{}) (resp interface{}, err error) {
+		if err = checkReq(grpcCheckRequest, k, req); err != nil {
+			return
+		}
+		resp, err = k.client.HealthClient.Check(ctx, req.(*grpc.CheckRequest))
+		return
+	}
+
 	k.reqHandlers[grpcExecProcessRequest] = func(ctx context.Context, req interface{}) (interface{}, error) {
+		if err := checkReq(grpcExecProcessRequest, k, req); err != nil {
```

Comment:
This will add 3 lines for each call, I'd suggest wrapping this like:

```golang
addHandler := func(handlerId string, k *kataAgent, f func(ctx context.Context, req interface{}) (interface{}, error)) {
	k.reqHandlers[handlerId] = func(ctx context.Context, req interface{}) (interface{}, error) {
		if err := checkReq(handlerId, k, req); err != nil {
			return nil, err
		}
		return f(ctx, req)
	}
}

addHandler(grpcCheckRequest, k, func(ctx context.Context, req interface{}) (interface{}, error) {
	return k.client.HealthClient.Check(ctx, req.(*grpc.CheckRequest))
})
addHandler(grpcExecProcessRequest, k, func(ctx context.Context, req interface{}) (interface{}, error) {
	return k.client.AgentServiceClient.ExecProcess(ctx, req.(*grpc.ExecProcessRequest))
})
```